qemu/tests/nic_teaming.py

```diff
@@ -0,0 +1,134 @@
+        status, output = session.cmd_status_output("ping %s -I team0 -c %s" 
+                                                   %(dest, count), timeout=600)
+        return status, output
+        #just for shotcut for following tests.
+        
+    def failover(ifnames):
+        time.sleep(3)
+        while True:
+            pid_ping = session_serial.cmd_output_safe("pgrep ping")
+            pattern_pid = r"(\d+)"
+            pid = re.findall(pattern_pid, pid_ping)
+            try:
+                pid[0]
+            except IndexError:
+                break
+                # if ping finished, will break the loop.
+            for port in ifnames:
+                session_serial.cmd_output_safe("ip link set %s down" %port)
+                time.sleep(random.randint(5, 30))
+                session_serial.cmd_output_safe("ip link set %s up" %port)
+                time.sleep(1)
+            # func for failover
+                
+    def check_ping(status, output):
+        if status != 0:
+            raise error.TestFail("Ping failed, staus:%s, output:%s"
+                                  % (status, output))
+        # if status != 0 the ping process seams hit issue.
+        ratio = utils_test.get_loss_ratio(output)
+        if ratio == -1:
+            raise error.TestFail('''The ratio is %s, and status is %s,
+                                    output is %s''' %(ratio,status,output))
+        elif ratio > 5:
+            raise error.TestFail("The loss raito is %s, test failed" %ratio)
+        logging.info("ping succeed with raito:%s" %ratio)
+        # ratio <5% is acceptance 
+                
+    error.context("Step1: Configure the team environment", logging.info)
+    #steps of building the teaming environment starts
+    modprobe_cmd = "modprobe team"
+    session_serial.cmd_output_safe(modprobe_cmd)
+    killteam0_cmd = "teamd -t team0 -k"
+    session_serial.cmd_output_safe(killteam0_cmd)
+    createteam_cmd = '''teamd -t team0 -d -c'''
+    createteam_cmd += ''' '{"runner" : {"name": "activebackup"}, '''
+    createteam_cmd += '''  "link_watch" : {"name": "ethtool"}}' '''
+    session_serial.cmd_output_safe(createteam_cmd)
+    # this cmd is to create the team0 and correspoding userspace daemon
+      
+    if  "Device not found" in session_serial.cmd_output_safe("ifconfig team0"):
+        raise error.TestFail("Interface team0 is not created.")
+    #check if team0 is created successfully.
+    print team_port_add()
+    print session_serial.cmd_output_safe("route -n")
+    #this is not this case checkpoint, just to check if route works fine
+    time.sleep(10)
+    # steps of building finished
+   
+          
+    try:
+        error.context("Login to guest", logging.info)
+        # steps of testing this case stats
+        session = vm.wait_for_login(timeout=timeout)
+        dest = utils_net.get_ip_address_by_interface(params.get("netdst"))
+        count = 100
+        print "IP address of host is %s" %dest
```

Comment:
this line will not recording into debug/info log, it's useless for debug in test loop